bookingsRoutes.js

```diff
@@ -0,0 +1,131 @@
+const express = require("express");
+const bookings = require("./bookings.json");
+const fs = require("fs").promises;
+
+const bookingsRouter = express.Router();
+
+
+// ----------- Getting All -----------
+const getAllBookings = (req, res) => {
+  fs.readFile("./bookings.json", "utf8")
+    .then(bookings => {
+      bookings = JSON.parse(bookings);
+      res.status(200).json({ data: bookings });
+      return;
+    });
+};
+
+
+// ----------- Posting -----------
+const postBooking = (req, res) => {
+  const { title, firstName, surname, email, checkInDate, checkOutDate } = req.body;
+  if (!title.trim() || !firstName.trim() || !surname.trim() || !email.trim() || !checkInDate.trim() || !checkOutDate.trim()) {
+    res.status(400).json("Please Fill all the Fields!");
+    return;
+  }
+
+  else {
+    const id = bookings[bookings.length - 1].id + 1;
+    const roomId = Math.floor(Math.random() * 10 + 1);
+    const newBooking = {
+      id, title, firstName, surname, email, roomId, checkInDate, checkOutDate
+    };
+
+    bookings.push(newBooking);
+
+    fs.writeFile("./bookings.json", JSON.stringify(bookings, null, 2), (err) => {
+      res.status(500).json({ message: "There is some Errors!" });
```

Comment:
It'd be good to log out what the error is here so you can debug it later, or send the error in the `res.json`.